### PR TITLE
Convert `BraintreePayPal_IntegrationTests`

### PR DESCRIPTION
**Pull request by mattwylder**
### Summary of changes

- Converted `BraintreePayPal_IntegrationTests` to Swift 
    - Fixed an issue where the vault flow was not actually tested in objective-c
- Converted `BTNonceValidationHelper` as `String+NonceValidation`
- `handleBrowserSwitchReturn` is now `internal`

### Checklist

~- [ ] Added a changelog entry~

- @jaxdesmarais @mattwylder

### CODE REVIEW

File: IntegrationTests/BraintreePayPal_IntegrationTests.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        ) { tokenizedPayPalAccount, error in
+            guard let nonce = tokenizedPayPalAccount?.nonce else {
+                XCTFail("Failed to tokenize account.")
+                return
+            }
+            XCTAssertTrue(nonce.isANonce)
+            XCTAssertNil(error)
+            tokenizationExpectation.fulfill()
+        }
+        self.waitForExpectations(timeout: 5)
```
Comment (jaxdesmarais) [inline IntegrationTests/BraintreePayPal_IntegrationTests.swift:44]:
```suggestion
        waitForExpectations(timeout: 5)
```

File: IntegrationTests/BraintreePayPal_IntegrationTests.swift [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+    
+    func testCheckoutFlow_withClientToken_tokenizesPayPalAccount() {
+        guard let apiClient = BTAPIClient(authorization: sandboxClientToken) else {
+            XCTFail("Failed to initialize BTAPIClient with sandbox tokenization key.")
+            return
+        }
+        let payPalClient = BTPayPalClient(apiClient: apiClient)
```
Comment (jaxdesmarais) [inline IntegrationTests/BraintreePayPal_IntegrationTests.swift:52]:
```suggestion
        }

        let payPalClient = BTPayPalClient(apiClient: apiClient)
```

File: IntegrationTests/BraintreePayPal_IntegrationTests.swift [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        let tokenizationExpectation = self.expectation(description: "Tokenize one-time payment")
```
Comment (jaxdesmarais) [inline IntegrationTests/BraintreePayPal_IntegrationTests.swift:53]:
```suggestion
        let tokenizationExpectation = expectation(description: "Tokenize one-time payment")
```

File: IntegrationTests/Helpers/String+NonceValidation.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+extension String {
+    var isANonce: Bool {
```
Comment (jaxdesmarais) [inline IntegrationTests/Helpers/String+NonceValidation.swift:4]: I think generally our style has been to have a space between extension and the first declaration - this is something that we can also handle with a linter in the future.

Comment (jaxdesmarais) [inline IntegrationTests/Helpers/String+NonceValidation.swift:4]: I wonder if `isAVaildNonce` or `isPaymentMethodNonce` or something similar would be clearer here. What are your thoughts?

Comment (mattwylder) [reply IntegrationTests/Helpers/String+NonceValidation.swift:4]: I like `isValidNonce`!

**Review by jaxdesmarais — APPROVED** [review]
👻

**Review by scannillo — APPROVED** [review]
🎸

**Pull request merged**